client.go

```diff
@@ -46,10 +46,16 @@ func NewWithClient(httpClient httputil.Client, serverName string) (*Client, erro
 	}
 
 	apiClient := &api.Client{
-		Client: httpClient,
+		Client:    httpClient,
+		Endpoints: api.Endpoints{Version: "r0"},
 	}
 	apiClient.HomeServer = parsed.Host
 	apiClient.HomeServerScheme = parsed.Scheme
+
+	if vClient, err := apiClient.WithLatestVersion(); err == nil {
```

Comment:
Would it be a better idea to report errors instead?

Reply:
I did this because `Discovery` borrows an initial client to check on `well_known`, but `WithLatestVersion` is called in `NewClient` instead. Calling `WithLatestVersion` doesn't actually make sense for that, so it's done optionally instead.